### DISCUSSION TITLE
Clipboard support

### DISCUSSION THREAD

**Discussion by fnc12** [axmolengine/axmol#2123] [General]
Is clipboard supported? I need to copy a string to clipboard but there is no crossplatform API for this in axmol found

**Comment by rh101** [comment]
> Is clipboard supported? I need to copy a string to clipboard but there is no crossplatform API for this in axmol found

It would be a little unreasonable to expect a game engine to have that kind of functionality, since it's not something that would typically be required in a game.

With a bit of research, it shouldn't be difficult to add the code in your own project to support that kind of functionality.  To get you started, here's an implementation that I personally use for Win32 projects:
```
void CopyToClipboard(const std::string& text)
{
    const auto hwnd = Director::getInstance()->getGLView()->getWin32Window();
    OpenClipboard(hwnd);
    EmptyClipboard();
    const auto hg = GlobalAlloc(GMEM_MOVEABLE, text.size() + 1);
    if (!hg) 
    {
        CloseClipboard();
        return;
    }
    memcpy(GlobalLock(hg), text.c_str(), text.size() + 1);
    GlobalUnlock(hg);
    SetClipboardData(CF_TEXT, hg);
    CloseClipboard();
    GlobalFree(hg);
}
```

**Reply by fnc12** [reply]
thanks for your quick reply. Yeah I found implementations for Windows and macOS. What I need it for Windows, Linux and macOS support. Of course I can implement it by myself. I was wondering if it exists already to copy and paste it without my custom implementation. Anyway thank you

**Comment by paulocoutinhox** [comment]
Try https://github.com/dacap/clip
